nasa/harmony-regression-tests
# DAS-1745  Build regression images on github and store on ghcr.io

This pr updates the harmony-regression-tests repo to build test images as
github actions.

Images are built with GitHub Actions, and then are pushed to ghcr.io.

It updates the scripts/* files to run the regression tests in AWS (from bamboo)
with  images built with the new Dockerfile.


## Jira Issue ID

[DAS-1745](https://bugs.earthdata.nasa.gov/browse/DAS-1745)


## Local Test Steps

Several test steps I think.

To test build of images: 

1. Create a fork of repo under your user. (uncheck fork main branch only so you include this one)
1. In your **forked repository**, create a PR for this branch and merge it branch to **yourrepository/main**.
1. Change the settings on your repository to allow write for workflow settings -> actions -> general -> workflow permissions read and write.
1. Ensure all images are built and pushed to ghrc.io 

You can look at my personal fork here: https://github.com/flamingbear/harmony-regression-tests/actions/

To Test utility of images:

1. Download all image, retag them to be the expected names. 

   ❯ docker pull ghcr.io/<username>/regression-tests-n2z:0.0.0
   ❯ docker tag ghcr.io/<username>/regression-tests-n2z:0.0.0 harmony/regression-tests-n2z
   
1. Run the regression tests.
   $ cd test
   $ ./run_notebooks.sh
   

To Test Bamboo will still function:

1. create EC2 keypair on SIT.
1. copy dot_env to .env fill out necessary environment variables. (this will
   print your AWS key and secret key to the screen, I am assuming they will not
   be printed in bamboo because of the way they are set, but should verify).
1. run tests on AWS.
   
   $ cd test
   $ ./test.sh 
   
All tests should run and pass.   


## Note

I found three possible problems with the tests running this.

1. Looks like swath-projector may be passing even with bad credentials. So I don't know if that's a necessary for the tests or if there's a problem with the tests.
1. Harmony regression tests isn't collecting tests. It passes every time.
1. If the regression image doesn't exist, the test pass.


   

## PR Acceptance Checklist
